### PR TITLE
Feature/react i18n update

### PR DESCRIPTION
**Pull request by emclaug2**
Changes proposed in this Pull Request:
- Convert example from JS to TS
- Convert React Class to FC
- Use React-i18n as the internationalization tool
- Add examples for Portuguese, Chinese, and French
- Add Lint and Prettier

### CODE REVIEW

**Comment by emclaug2** [comment]
I'm going to spend a bit of time adding some better unit tests for this.

**Comment by huayunh** [comment]
snack bar needs a `box-sizing: border-box`

**Comment by emclaug2** [comment]
> * Refreshing the application does not reset the language to English
> * There are still some console logs and a console warning
> 
> Potentially out of scope, but might be worth doing if we're already in here refactoring the example:
> 
> * Would be nice to fix the keyline alignment of the menu button (16px gap to the left, 32px to the right)
> * Can we use our InfoListItem component and our Drawer component?

I updated the example to use the InfoListItem.

I started converting the example to use our Drawer, but then realized if we move this example over to the mono repo, then we'd have a drawer within a drawer.  I think it's best if we just remove the Drawer from this example since it doesn't really add anything in terms of showing off how to do Internationalization.  This is not a navigation pattern.   What do you think?

**Comment by joebochill** [comment]
Updates look good. Are we going to remove the drawer?

**Comment by emclaug2** [comment]
> Updates look good. Are we going to remove the drawer?

I'm going to leave the original drawer for now seeing that when we move this code over to the mono repo we might have to remove the drawer.

**Comment by jeffvg** [comment]
I see warning on deprecation...
- Warning: React.createFactory() is deprecated and will be removed in a future major release. Consider using JSX or use React.createElement() directly instead

**Comment by emclaug2** [comment]
> I see warning on deprecation...
> 
> * Warning: React.createFactory() is deprecated and will be removed in a future major release. Consider using JSX or use React.createElement() directly instead

I'm not sure what's causing this warning because there's no stacktrace to go off of.  I think it might have something to do with the i18n stuff, but I'd have to remove it all to test.

**Comment by jeffvg** [comment]
Does the 400 font weight on Arabic seem hard to see or is it me

**Comment by jeffvg** [comment]
Odd menu behavior when switching from English to any other language in menu causes menu to shift up.
menu-int

**Comment by emclaug2** [comment]
> Odd menu behavior when switching from English to any other language in menu causes menu to shift up.
> menu-int

This is the default behavior of the mat-ui select form.  Sometimes it will open the list relative to your selection current selection.  Here's an example from their docs: https://1ucesz.run.stackblitz.io

File: package.json [hunk 1]
```diff
@@ -3,28 +3,46 @@
   "version": "0.0.0",
   "private": true,
   "dependencies": {
-    "@material-ui/core": "3.9.3",
+    "@material-ui/core": "^4.0.0",
     "@material-ui/icons": "3.0.2",
```
Comment (joebochill) [inline package.json:7]: update this to ^4.0.0 to eliminate the createFactory warning

File: src/resources/i18n.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+            en: english,
+            ar: arabic,
+            es: spanish,
+            de: german,
+            fr: french,
+            pt: portuguese,
+            zh: chinese,
+        },
+        fallbackLng: 'english',
+        debug: true,
```
Comment (joebochill) [inline src/resources/i18n.tsx:27]: set this to false to disable the console logs

File: package.json (hunk 1 above)
Comment (emclaug2) [reply package.json]: Awesome info! Updated.

File: src/resources/i18n.tsx (hunk 2 above)
Comment (emclaug2) [reply src/resources/i18n.tsx]: Good catch, set this to false.

**Pull request merged**